x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/factories/utils/build_bulk_body.ts

```diff
@@ -35,19 +36,35 @@ export const buildBulkBody = (
   ruleSO: SavedObject<AlertAttributes>,
   doc: SignalSourceHit,
   mergeStrategy: ConfigType['alertMergeStrategy'],
-  applyOverrides: boolean
+  applyOverrides: boolean,
+  buildReasonMessage: BuildReasonMessage
 ): RACAlert => {
   const mergedDoc = getMergeStrategy(mergeStrategy)({ doc });
   const rule = applyOverrides
     ? buildRuleWithOverrides(ruleSO, mergedDoc._source ?? {})
     : buildRuleWithoutOverrides(ruleSO);
   const filteredSource = filterSource(mergedDoc);
+  const timestamp = new Date().toISOString();
+  let hostName = '';
+  let userName = '';
+  if (mergedDoc.fields) {
+    hostName = mergedDoc.fields['host.name'] != null ? mergedDoc.fields['host.name'] : hostName;
+    userName = mergedDoc.fields['user.name'] != null ? mergedDoc.fields['user.name'] : userName;
+  }
+  const reason = buildReasonMessage({
+    alertName: rule.name,
+    alertRiskScore: ruleSO.attributes.params.riskScore,
```

Comment:
riskScore and severity can be overridden in `buildRuleWithOverrides` - is this specifically intending to use the static values from the rule SO? I would guess if the value gets overridden we want to use that in the reason.

Reply:
Thanks, updated!